homeassistant/components/lamarzocco/number.py

```diff
@@ -92,6 +110,97 @@ class LaMarzoccoNumberEntityDescription(
 )
 
 
+KEY_ENTITIES: tuple[LaMarzoccoKeyNumberEntityDescription, ...] = (
+    LaMarzoccoKeyNumberEntityDescription(
+        key="prebrew_off",
+        translation_key="prebrew_off",
+        icon="mdi:water-off",
+        device_class=NumberDeviceClass.DURATION,
+        native_unit_of_measurement=UnitOfTime.SECONDS,
+        native_step=PRECISION_TENTHS,
+        native_min_value=1,
+        native_max_value=10,
+        entity_category=EntityCategory.CONFIG,
+        set_value_fn=lambda lm, off_time, key: lm.configure_prebrew(
+            on_time=int(lm.current_status[f"prebrewing_ton_k{key}"] * 1000),
+            off_time=int(off_time * 1000),
+            key=key,
+        ),
+        native_value_fn=lambda lm, key: lm.current_status[f"prebrewing_ton_k{key}"],
+        enabled_fn=lambda lm: lm.current_status["enable_prebrewing"],
+        not_settable_reason="Prebrewing is not enabled",
+        supported_fn=lambda coordinator: coordinator.lm.model_name
+        in (
+            LaMarzoccoModel.LINEA_MICRA,
+            LaMarzoccoModel.LINEA_MINI,
+            LaMarzoccoModel.GS3_AV,
+        ),
```

Comment:
Since we use this set 3 times, I think its more performant to have this as separate variable outside

Reply:
I could also just check for ` != LaMarzoccoModel.GS3_MP` if you prefer that